h/static/scripts/annotation-ui.js

```diff
@@ -16,59 +19,106 @@ function initialSelection(settings) {
   return value(selection);
 }
 
+function initialState(settings) {
+  return Object.freeze({
+    visibleHighlights: false,
+
+    // Contains a map of annotation tag:true pairs.
+    focusedAnnotationMap: null,
+
+    // Contains a map of annotation id:true pairs.
+    selectedAnnotationMap: initialSelection(settings),
+  });
+}
+
+var types = {
+  SELECT_ANNOTATIONS: 'SELECT_ANNOTATIONS',
+  FOCUS_ANNOTATIONS: 'FOCUS_ANNOTATIONS',
+  SET_HIGHLIGHTS_VISIBLE: 'SET_HIGHLIGHTS_VISIBLE',
+};
+
+function reducer(state, action) {
```

Comment:
If you're not already familiar with redux, the purpose of this function is completely opaque.

A short paragraph at the top of the file explaining the purpose of the module, and the basic approach to its design (including explaining how redux provides "encapsulated state") would go a long way to making this code more transparent.